source/rust_verify/src/rust_to_vir_func.rs

```diff
@@ -519,6 +700,91 @@ fn is_mut_ty<'tcx>(
     }
 }
 
+fn predicates_match<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    id1: rustc_span::def_id::DefId,
+    id2: rustc_span::def_id::DefId,
+    substs: SubstsRef<'tcx>,
+) -> bool {
+    let preds1 = all_predicates(tcx, id1, substs);
+    let preds2 = all_predicates(tcx, id2, substs);
+    if preds1.len() != preds2.len() {
+        return false;
+    }
+    // TODO should allow this to work if the predicates are in a different order
```

Comment:
With the code as-is, what happens if they aren't?

Reply:
there will be an error about trait mismatch